theme/static/js/hs_file_browser.js

```diff
@@ -919,14 +919,28 @@ $(document).ready(function () {
         });
     });
 
-     // Download method
+    // Download method
     $("#btn-download, #fb-download").click(function () {
         var downloadList = $("#fb-files-container li.ui-selected");
+
+        // Remove previous temporary download frames
+        $(".temp-download-frame").remove();
+
         if (downloadList.length) {
-            for (var i = 0; i < downloadList.length; i++) {
-                var url = $(downloadList[i]).attr("data-url");
-                var fileName = $(downloadList[i]).children(".fb-file-name").text();
-                downloadURI(url, fileName);
+            // Workaround for Firefox and IE
+            if (jQuery.browser.mozilla == true || !!navigator.userAgent.match(/Trident\/7\./)) {
```

Comment:
@mauriel133 I meant whether double ```!``` should be replaced with single ```!``` initially. Now that I think about it, wouldn't double ```!``` be equivalent to no ```!``` at all? In other words, if it matches, it means IE browser, then the condition is true, and if it does not match, it is null, then the condition is false. Not sure whether double ```!``` is needed here.